projects/rulesets-base/src/extended-rules/spectral-rule.ts

```diff
@@ -145,3 +161,50 @@ export class SpectralRule extends ExternalRuleBase {
     return results;
   }
 }
+
+async function runSpectral(
+  spectralRuleset: string,
+  afterSpecAbsolutePath: string
+): Promise<string> {
+  const resultsOutput = path.join(
+    os.tmpdir(),
+    `spectral-output-${Math.floor(Math.random() * 100000)}.json`
```

Comment:
a `nanoid` or `uuid` might be more appropriate here, but this should also be fine (cant remember which one we have in rulesets base but we have at least one of them)